clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/utils/KubeTestUtils.java

```diff
@@ -255,14 +253,102 @@ public static void disableManifest(
                           + (targetNs.isBlank() ? "''" : targetNs)
                           + ".flatten()",
                       String.class));
-          return respTask.jsonPath().getBoolean("status.completed");
+        });
+
+    return deployedObjectNames;
+  }
+
+  public static String sendOperationExpectFailure(
+      String baseUrl,
+      List<Map<String, Object>> reqBody,
+      String targetNs,
+      long duration,
+      TimeUnit unit)
+      throws InterruptedException {
+    String taskId = submitOperation(baseUrl, reqBody);
+
+    // The last status is most interesting, but a single String isn't
+    // effectively final and so can't be used in a lambda.  So, use a list.  Any
+    // kind of wrapper object would do.
+    List<String> status = new ArrayList<>();
+    processOperation(
+        baseUrl,
+        taskId,
+        duration,
+        unit,
+        (Response respTask) -> {
+          respTask.then().body("status.failed", is(true));
+          status.add(respTask.jsonPath().getString("status.status"));
+        });
+
+    // Return the status to the caller so it's possible to assert on it.
+    return Iterables.getOnlyElement(status);
+  }
+
+  /** Wait until a task has completed, calling a consumer with the response once it has. */
+  private static void processOperation(
+      String baseUrl, String taskId, long duration, TimeUnit unit, Consumer<Response> consumer)
+      throws InterruptedException {
+    System.out.println("> Waiting for task to complete");
+    long start = System.currentTimeMillis();
+
+    KubeTestUtils.repeatUntilTrue(
+        () -> {
+          Response respTask = getTask(baseUrl, taskId);
+          if (respTask == null) {
+            return false;
+          }
+          consumer.accept(respTask);
```

Comment:
This could be moved outside the `repeatUntilTrue` loop, just in case the `accept` method throws an exception (like when a test case fails) it gets propagated more cleanly.

Reply:
Like this? https://github.com/spinnaker/clouddriver/pull/5678/commits/f8896468d42d28d4069787656f88068afe8e4ce0